src/react-native/04-intro-to-jsx/intro-to-jsx.md

```diff
@@ -201,9 +201,215 @@ Next, we want to render a list of states name in our application:
 
-### Concept TODO
+To do so, we‘ll learn about:
 
-TODO
+- Using JavaScript variables and expressions in JSX
+- Working with conditionals and loops in JSX
+
+### Using JavaScript variables and expressions in JSX
+
+JSX is dynamic. You can insert values from variables and objects into your JSX as we did with the image URL in the previous section.
+
+```tsx
+const name = "Bitovi"
+
+const content = <Text>Welcome to {name}!</Text>
+```
+
+In the code above, use the `{name}` syntax to tell JSX to render the value stored in the `name` variable (i.e. `"Bitovi"`) into our view.
+
+You can take this a step further by interpolating multiple values and using JavaScript functions to transform data on the fly. Anything that goes inside `{}` is executed as normal JavaScript. These are the same rules as the brackets on a prop: any JavaScript expression is valid inside the curly brackets.
+
+```tsx
+const person = {
+  name: "mike",
+  profession: "programmer",
+}
+
+const content = (
+  <View>
+    <Text>Hi I’m {person.name.toUpperCase()}!</Text>
+    <Text>I’m a {person.profession} living in Philadelphia.</Text>
+  </View>
+)
+```
+@highlight 8,9
+
+### JSX is a syntax extension for JavaScript
+
+Remember, JSX is an alternative syntax for normal JavaScript—it is not magic. This means that you can use JSX as a normal value, too.
+
+```tsx
+const header = <Text>Hello World</Text>
+const body = <Text>My name is {"Mike"}</Text>
+
+function MyPage() {
+  return (
+    <View>
+      {header}
+      {body}
+    </View>
+  )
+}
+```
+
+If rendered, the `screen` will output:
+
+```xml
+<View>
+  <Text>Hello World</Text>
+  <Text>My name is Mike</Text>
+</View>
+```
+
+If this surprises you, remember that underneath the syntactic sugar, JSX is nothing more than `React.createElement` calls:
+
+```tsx
+const header = React.createElement("Text", null, "Hello World")
+const body = React.createElement("Text", null, `My name is ${"Mike"}`)
+
+const page = React.createElement("View", null, [header, body])
+```
+
+### Working with conditionals and loops in JSX
+
+Only expressions that return a value may be interpolated. This includes static values, variables and calls to functions. It does not include control-flow statements such as `if`, `case`, `for`, `while`. These can either be abstracted behind a function, which is then called within the JSX or be re-written in a JSX-friendly way.
+
+To put it simply: only things that you could pass into a function can be used inside the brackets.
+
+#### Using conditionals
+
+Conditions can be rewritten using the ternary operator.
+
+```tsx
+// This does not work
+const content = (
+  <Text>
+    {
+      if (a === b) { // Control flow does not belong in JSX
+        "a and b are equal"
+      } else {
+        "a and b are different"
+      }
+    }
+  </Text>
+)
+```
+
+```tsx
+// But the same can be accomplished with ternaries
```

Comment:
The comments at the top of these examples should be moved into the main paragraph text.